File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30StrategyJavaScript/controller/SortController.js
import { BubbleSort } from './BubbleSort.js';
import { InsertionSort } from './InsertionSort.js';
import { QuickSort } from './QuickSort.js';
import { SortingContext } from './SortingContext.js';

export class SortController {
    constructor(model, view) {
        this.model = model;
        this.view = view;
        this.context = new SortingContext();
    }

    sortNumbers() {
        const numbers = this.model.getNumbers();
        if (numbers.length <= 30) {
            this.context.setSortStrategy(new BubbleSort());
        } else if (numbers.length <= 100) {
            this.context.setSortStrategy(new InsertionSort());
        } else {
            this.context.setSortStrategy(new QuickSort());
        }

        this.context.sort(numbers);
        this.view.displaySortedNumbers(numbers, this.context.getSortStrategyName());
    }
}

File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30StrategyJava/src/main/java/ec/edu/espe/controller/SortController.java

package ec.edu.espe.controller;

import ec.edu.espe.model.SortModel;
import ec.edu.espe.view.SortView;

/**
 *
 * @author MSI-PULSE
 */
public class SortController {
    private SortModel model;
    private SortView view;
    private SortingContext context;

    public SortController(SortModel model, SortView view) {
        this.model = model;
        this.view = view;
        this.context = new SortingContext();
    }

    public void sortNumbers() {
        int[] numbers = model.getNumbers();
        if (numbers.length <= 30) {
            context.setSortStrategy(new BubbleSort());
        } else if (numbers.length <= 100) {
            context.setSortStrategy(new InsertionSort());
        } else {
            context.setSortStrategy(new QuickSort());
        }

        context.sort(numbers);
        view.displaySortedNumbers(numbers, context.getSortStrategyName());
    }
}